cvat/apps/engine/backup.py

```diff
@@ -331,19 +334,35 @@ def serialize_segment(db_segment):
             segment = segment_serailizer.data
             segment.update(job_data)
 
+            if self._db_data.custom_segments:
+                segment.update(serialize_custom_file_mapping(db_segment))
+
             return segment
 
         def serialize_jobs():
             db_segments = list(self._db_task.segment_set.all())
             db_segments.sort(key=lambda i: i.job_set.first().id)
             return (serialize_segment(s) for s in db_segments)
 
+        def serialize_custom_file_mapping(db_segment: models.Segment):
+            if self._db_task.mode == 'annotation':
+                files: Iterable[models.Image] = self._db_data.images.all().order_by('frame')
+                segment_files = files[db_segment.start_frame : db_segment.stop_frame + 1]
+                return {'files': list(frame.path for frame in segment_files)}
```

Comment:
Backups have such manifest only in some cases. Also, it will be simpler to extend this parameter in future if it is saved separately. For instance, if jobs can contain arbitrary files or have overlaps.